src/components/enter-authenticator-app-code/index.njk

```diff
@@ -40,7 +40,6 @@
 
   </form>
 
-  {% if isAccountRecoveryPermitted === true %}
```

Comment:
Hmm, I'm not sure about this. Presumably there's a reason for this line outside of the case you're looking at. Are we sure this can be totally discarded (as opposed to tweaking something for the thing we're trying to fix)

Reply:
From what I managed to understand, this should be ok. When the user is required a higher level of trust another template is being rendered "index-2fa-service-uplift-auth-app.njk". There the link will still not be visible after a wrong code is submitted in those cases.

Reply:
Cool, I'll let others wade in on this (I see @gtvj did some of these changes). But this kind of context is really important to put in commit messages if you've found out why something that looks like it's needed isn't needed

Reply:
Some of the smallest changes have the most context to them - see https://dhwthompson.com/2019/my-favourite-git-commit for a fun example

Reply:
Hello @VladGavrilet @BeckaL 👋🏽 

I've just been looking back through the code and it seems the original condition was introduced in [the PR](https://github.com/govuk-one-login/authentication-frontend/pull/964) for AUT-1088 to link the availability of the `<details>` element to account recovery being enabled for “the environment and the user”. I can see from that initial PR that:

* both `req.session.user.isAccountRecoveryPermitted` and `isAccountRecoveryPermittedForUser` are assigned to what we get back from the `accountRecoveryResponse` (See [here](https://github.com/govuk-one-login/authentication-frontend/blob/e0174382775969c4a09e0600c627538f0ddd4898/src/components/enter-authenticator-app-code/enter-authenticator-app-code-controller.ts#L80))
* `isAccountRecoveryPermittedForUser ` is what we pass through to the template as `isAccountRecoveryPermitted` (See [here](https://github.com/govuk-one-login/authentication-frontend/blob/e0174382775969c4a09e0600c627538f0ddd4898/src/components/enter-authenticator-app-code/enter-authenticator-app-code-controller.ts#L91)) 
* the API handler can return false for `accountRecoveryPermitted` (See [here](https://github.com/govuk-one-login/authentication-api/blob/bf8b6733f5fdef4d5724f1ec5e4596fd3e957864/frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/AccountRecoveryHandler.java#L95) - sorry in advance if I’ve misunderstood this, I'm not the best at interpreting Java code)

If that’s the case, it does seem there is a case where the `{% if isAccountRecoveryPermitted === true %}` condition would still need to apply, in which case I think we might need a different approach. 

I actually wonder if the problem here is that `accountRecoveryPermitted` isn’t being carried through to the template following a validation error (have a vague memory that we might have seen this kind of thing before). 

Sorry this took me a while to interpret - I think my only involvement in this before was introducing a [strict equality check](https://github.com/govuk-one-login/authentication-frontend/pull/1109). Looking at that PR the bug I was fixing arose from users being presented with the link when it shouldn't have been available to them. 

I feel like it might be good to include some of this context in the updated PR and commit message.